httpserver-rs/tests/httpserver_test.rs

```diff
@@ -0,0 +1,231 @@
+//! unit test of httpserver
+//!
+//! This loads the httpserver provider, links it to a mock actor (below),
+//! and issues a GET and a PUT request. The actor receives the request and responds
+//! with json summary of the request. (similar to what the example Echo actor does)
+//!
+//! If the test fails with an error
+//!    error creating server listener: Address already in use
+//! it's because the listening port (set on provider_test_config.toml) is in use.
+//! If the only listener on that port is this test, you may have another test process
+//! running because a previous run didn't shut down cleanly
+//! (this can happen if it failed failed with a panic error).
+//! If you're on linux or macos:
+//!     Check the output of `ps ax | grep httpserver`,
+//!     If it has one or more processes called 'target/debug/httpserver', they're from this test.
+//!     Try `killall httpserver` to kill them.
+//!
+use wasmbus_rpc::core::InvocationResponse;
+use wasmbus_rpc::provider::prelude::*;
+use wasmcloud_interface_httpserver::*;
+use wasmcloud_test_util::{
+    check,
+    cli::print_test_results,
+    provider_test::test_provider,
+    testing::{TestOptions, TestResult},
+};
+#[allow(unused_imports)]
+use wasmcloud_test_util::{run_selected, run_selected_spawn};
+
+/// HTTP host and port for this test.
+/// Port number should match value in provider_test_config.toml
+const SERVER_UNDER_TEST: &str = "http://localhost:9000";
```

Comment:
I'm a fan of not putting this on `8080` as that drastically lowers the chance of accidental duplication with our examples!